### ISSUE TITLE
Change the HTTP Request default span name

### ISSUE THREAD

**Issue by aguimaraes** [elastic/apm-agent-php#993]
**Is your feature request related to a problem? Please describe.**
Most of our HTTP requests coming from outside the perimeter are going through an API Gateway.
Since the default HTTP request instrumentation uses the schema `<method> <host>` to name spans, we end up with multiple calls like `GET api.example.com` without being able to easily differentiate between the different endpoints we are calling.
When we need to quickly identify a problem with individual calls, it becomes hard exercise quickly.

**Describe the solution you'd like**
I would like to be able to change the default span name for HTTP requests on initialisation time using an environment variable or a php.ini param.

**Describe alternatives you've considered**
We could also change the default schema to use `<method> <path>` instead and we would start to see:
`GET api.example.com/user/333`.

**Additional context**
Current Spec suggests the same naming schema: https://github.com/elastic/apm/blob/main/specs/agents/tracing-instrumentation-http.md

**Comment by SergeyKleyman** [comment]
Hi @aguimaraes 

For now we don't have such feature planned near term but as a potential workaround we can propose [transforming data with custom ingest pipelines](https://www.elastic.co/guide/en/fleet/8.11/data-streams-pipeline-tutorial.html). Spans for outgoing HTTP requests have [URL in their context](https://github.com/elastic/apm-server/blob/7.0/docs/spec/spans/span.json#L73) so it can be used to create a new name.

In the future we plan to allow registering a span processors (as PHP callable-s) at agent level that will allow modifying spans before they sent out - will this approach be useful to you?

**Comment by aguimaraes** [comment]
yes @SergeyKleyman, the future approach would work perfectly for our use case.
Thanks for the workaround suggestion too!
Have a great day!
